docs/synapse/devopsguide.rst

```diff
@@ -930,6 +930,621 @@ not be emitted.
+or if the endpoints' View is deleted from the Cortex, this will raise a fatal error and return an HTTP 500 error. Once
+a query has started executing, regular Storm permissions apply.
+
+Endpoints can also have permissions defined for them. This allows locking down an endpoint such that while a user may
+still have access to the underlying view, they may lack the specific permissions required to execute the endpoint.
+These permissions are checked against the authenticated user, and not the endpoint owner. The following example shows
+setting a single permission on a one of our earlier endpoints::
+
+    $api=$lib.cortex.httpapi.get(bd4679ab8e8a1fbc030b46e275ddba96)
+    $api.perms=(your.custom.permission,)
+
+Accessing it as a unpriveleged user generates an ``AuthDeny`` error::
+
+    $ curl -sku "lowuser:demo" "https://127.0.0.1:4443/api/ext/demo/owner"  | jq
+    {
+      "status": "err",
+      "code": "AuthDeny",
+      "mesg": "User (lowuser) must have permission your.custom.permission"
+    }
+
+The user can have that permission granted via Storm::
+
+    storm> auth.user.addrule lowuser your.custom.permission
+    Added rule your.custom.permission to user lowuser.
+
+Then the endpoint can be accessed::
+
+    $ curl -sku "lowuser:demo" "https://127.0.0.1:4443/api/ext/demo/owner"  | jq
+    "root"
+
+For additional information about managing user permissions, see :ref:`admin_create_users_roles`.
+
+.. note::
+
+    When the Optic UI is used to proxy the ``/api/ext`` endpoint, authentication must be done using the Optic's login
+    endpoint. Basic auth is not available.
+
+Readonly Mode
++++++++++++++
+
+The Storm queries for a given handler may be executed in a ``readonly`` runtime. This is disabled by default. This can
+be changed by setting the ``readonly`` attribute on the ``http:api`` object::
+
+    // Enable the Storme queries to be readonly
+    $api = $lib.cortex.httpapi.get($yourIden)
+    $api.readonly = $lib.true
+
+Endpoint Variables
+++++++++++++++++++
+
+User defined variables may be set for the queries as well. These variables are mapped into the runtime for each method.
+This can be used to provide constants or other information which may change, without needing to alter the underlying
+Storm code which defines a method. These can be read ( or removed ) by altering the ``$api.vars`` dictionary. This is
+an example of using variable in a query::
+
+    // Set a variable that a method uses:
+
+    $api = $lib.cortex.httpapi.get($yourIden)
+    $api.methods.get = ${
+        $mesg = `There are {$number} things available!`
+        $request.reply(200, body=({"mesg": $mesg})
+    }
+    $api.vars.number = (5)
+
+When executing this method, the JSON response would be the following::
+
+    {"mesg": "There are 5 things available!"}
+
+If ``$api.vars.number = "several"`` was executed, this would be the following output::
```

Comment:
"this would be the following output" reads a bit odd to me, could probably just have it be "the JSON response would be the following" like above